### ISSUE TITLE
Escape double quote character in double quoted string

### ISSUE THREAD

**Issue by haint504** [vlang/v#2216]
**V version:** 1.3.0
**OS:** MacOS

**What did you do?**
```
fn main() {
    a := 10 
    println("\"$a")
}
```

Compiler error: 
```
error: use of undeclared identifier 'd'
printf( "\\"%d\n", a ) ;
             ^
```

**What did you expect to see?**
Compile successfully and run with output `"10`
 
**What did you see instead?**

**Comment by tomByrer** [comment]
VLang can [output C source](https://github.com/vlang/v/wiki/Troubleshooting).   Have you tried to compile the C code to WASM?

https://dassur.ma/things/c-to-webassembly/
https://developer.mozilla.org/en-US/docs/WebAssembly/C_to_wasm

**Comment by tomByrer** [comment]
> println("\"$a')

Seems to me you mismatched quote strings here.

TBH, I knew allowing 2 types of quoting would be trouble.

**Comment by haint504** [comment]
True, the correct example which is a bug of compiler is 
```
fn main() {
    a := 10 
    println("\"$a")
}
```

Thanks for correcting. And yes, I would go against allowing 2 types of quoting.

**Comment by gslicer** [comment]
I think the reason @medvednikov preferred `'` as first was that you are able to enter it without bothering woth `SHIFT` or `ALT` - if I remember correctly the auto-code-formatter will convert all `"` enclosed strings to `'` (and then only when no `'` is occurring - which I don't understand, as the parser could easily escape them and keep all string consistent...)

**Comment by oleg-kachan** [comment]
> Thanks for correcting. And yes, I would go against allowing 2 types of quoting.

AFAIK from version 0.1.21 we have double quoting too,

I also would prefer only one type of quotes - doubles

**Comment by Henrixounez** [comment]
You don't use double quotes for strings in V.
The way you should do you example is :
```
fn main() {
        a := 10
        println('"$a')
}
```
It is true however that a warning for using double quotes instead of single ones should be prompted.

**Comment by haint504** [comment]
> You should prefer using single quotes for strings in V.
> The way you should do you example is :
> 
> ```
> fn main() {
>         a := 10
>         println('"$a')
> }
> ```
> 
> I kind of prefered to only using single quotes, no confusion.

It's not about preference, if V accepts double quote but shows compile error for example above, clearly it's compiler bug. Agree we should embrace single quote only.

**Comment by bogen85** [comment]
yeah, but multiple quoting types breaks down when you need to use all of them in the string.
Some sort of escaping is needed unless you have raw strings.
It is escaping a quote in general that is not allowed in V.
```
V 0.1.23 68b4dab
Use Ctrl-C or `exit` to exit
>>> a := "foo"
>>> b := "foo\"bar"
/tmp/v/.vrepl_temp.tmp.c: In function ‘main’:
/tmp/v/.vrepl_temp.tmp.c:3257:24: error: expected ‘)’ before ‘bar’
 3257 |  string b= tos3("foo\\"bar") ;
      |                        ^~~
      |      ...
(Use `v -g` to print the entire error message)
V error: C error. This should never happen. 
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
>>>
```

```
>>> c := "foo\cbar"
>>> c
foocbar
>>>
```
The documentation does not mention escaping quotes.
https://vlang.io/docs#strings

However, with string interpolation doing a lot of escaping is not as needed.
```
$ v
V 0.1.23 68b4dab
Use Ctrl-C or `exit` to exit
>>> q := '"'
>>> d := 'dog'
>>> a := '$q$d$q'
>>> a
"dog"
>>>  d
dog
>>>
```

**Comment by orneo1212** [comment]
Fixed by #6474

**Comment by tokamak-git** [comment]
Hey folks has this been solved?

I'm facing a similar error. Am using regex for some pattern matching and find that I'm unable to escape a '

```
fn is_valid_pattern_upper(input string) bool {
  // query := r'^[A-Z0-9\s!@#$%^&*()_+\-=\[\]{};\':"\\|,.<>/?]+\?$'
  query := r'^[A-Z0-9\s!@#$%^&*()_+\-=\[\]{};:"\\|,.<>/?]+$'
  mut re := regex.regex_opt(query) or { panic(err) }
  start, end := re.match_string(input)
  return start >= 0 && end == input.len
}
```

Here when I use the commented out query I get a compile error.
